rsciio/topspin/_api.py

```diff
@@ -0,0 +1,428 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2025 The HyperSpy developers
+#
+# This file is part of RosettaSciIO.
+#
+# RosettaSciIO is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# RosettaSciIO is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with RosettaSciIO. If not, see <https://www.gnu.org/licenses/#GPL>.
+
+
+import datetime
+import os
+import sys
+import xml.etree.cElementTree as ET
+
+import h5py
+import numpy as np
+from tqdm import tqdm
+
+from rsciio._docstrings import FILENAME_DOC, RETURNS_DOC
+
+
+def file_reader(filename, subset=None, dryrun=False, show_progressbar=True):
+    """
+    Read .app5 file format used by NanoMegas's Topspin software.
+
+    .app5 files use the hdf5 file format, with metadata stored as
+    binarized XML-style text strings.
+
+
+    Parameters
+    ----------
+    %s
+    subset: str or None
+        h5py-style address. If given, only the subset of experiments
+        located at this address will be imported. If none, all datasets
+        will be imported.
+
+    dryrun : bool
+        If True, the .app5 files are quickly scanned without being loaded,
+        and a summary is printed to the log. Default is False.
+
+    show_progressbar: bool, default=True
+        Whether to show the progressbar or not. If True, shows progress
+        bar for each individual dataset loaded from the original file.
+    %s
+
+    Notes
+    -----
+    The hierarchy of the Metadata objects stored in app5 files changes
+    based on Topspin Proceedure and local microscope setup. RosettaSciIo
+    is still able to import datsets without this information, but be aware
+    that for new or customized setups some metadata entrees might be missing.
```

Comment:
In this case, the user should be made aware that some metadata are missing maybe by using the logger with level info or warning depending on criticality of this missing information? Document the behaviour here.